solutions_builder/cli/vars.py

```diff
@@ -0,0 +1,138 @@
+"""
+Copyright 2023 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import typer
+import pathlib
+import re
+import jinja2
+from typing import Optional
+from typing_extensions import Annotated
+from copier import run_auto
+from .cli_utils import *
+
+vars_app = typer.Typer()
+
+INCLUDE_PATTERNS = [
+    "*.yaml", "*.yml", "*.env", "*.tfvars", "*.tf", "*.sh", "*.md"
+]
+EXCLUDE_PATTERNS = ["**/.terraform/**/*.*", "**/node_modules/**/*.*", "**/.venv/**/*.*"]
+
+# Replace a varialbe with a given text content.
+def replace_var_to_template(var_name, text, custom_template=False, debug=False):
+  match_pattern = f"^([^\\r]*[:|=]\\s*)([\"\']?)([^\"^\']*)([\"\']?)\\s*#\\s*sb-var:{var_name}"
```

Comment:
Please add a comment on what match pattern is trying to do (for future developers)

Reply:
added comments.